app/helpers/catalog_helper.rb

```diff
@@ -53,7 +54,7 @@ def service_catalog_summary(record, sb_data)
       data[:rows].push(row_data(_('Price / Month (in %{currency})') % {:currency => record.currency.code}, record.price.to_s))
     end
     disable = !record.template_valid?
-    action = disable ? '' : "miqOrderService(#{record.id});"
+    action = disable ? {} : {:recordId => record.id, :type => 'ORDER-SERVICE'}
```

Comment:
For example here, can the action be a function?

```suggestion
    action = () => { if (!disable) { miqOrderService(record.id) }} }
```

Reply:
` action = () => { if (!disable) { miqOrderService(record.id) }} } `
looks more like a mix of rails and javascript syntax?
because this is a helper file and it would look for a method named `miqOrderService`

this would work in javascript though
![image](https://github.com/ManageIQ/manageiq-ui-classic/assets/87487049/3ab29331-f6d4-45f7-8cc2-1f0d1fac7b11)

Reply:
I don't know how to javascript - that was basically psuedocode 😅

Reply:
ohhhhhh - i was confused because I thought all of these files were js/jsx - I didn't realize these changes were on the _Ruby_ side.  Now I see what you're doing is sending over metadata to the javascript side so that the javascript side can construct callbacks but without using eval. 💡